hail/src/test/scala/is/hail/expr/ir/EmitStreamSuite.scala

```diff
@@ -193,6 +193,38 @@ class EmitStreamSuite extends HailSuite {
     assert(range(null) == null)
   }
 
+  @Test def testEmitSeqSample(): Unit = {
+    val N = 20
+    val n = 2
+
+    val seqIr = SeqSample(
+        I32(N),
+        I32(n),
+        false
+      )
+
+    val compiled = compileStream[AsmFunction1RegionLong, Unit](seqIr, FastIndexedSeq()) { (f, r, _) => f(r) }
+
+    // Generate many pairs of numbers between 0 and N, every pair should be equally likely
+    val results = Array.tabulate(N, N){ case(i, j) => 0}
+    (0 until 1000000).foreach { i =>
+      val seq = compiled.apply(()).map(_.asInstanceOf[Int])
+      assert(seq.size == n)
+      results(seq(0))(seq(1)) += 1
+      assert(seq.toSet.size == n)
+      assert(seq.forall(e => e >= 0 && e < N))
+    }
+
+
+    (0 until N).foreach { i =>
+      (i + 1 until N).foreach { j =>
+        val entry = results(i)(j)
+        // Expected value of entry is 5000
+        assert(entry > 4000 && entry < 6000)
```

Comment:
It's safe to tighten this up. I'd suggest 4880 and 5650. The probability that not all entries are within that window is < 1e-5. (Expected value is 5263.)

Reply:
Done, thanks. I mistakenly thought it was 5000.